add authorisation jwt-tokens and move configuration to appsetings.json

zloypelmen opened the pull request.

zloypelmen committed:
add something big (DbContext, Identity)

zloypelmen committed:
move configuration to appsetings.json

zloypelmen committed:
Merge pull request #4 from JenBytowski/feature/identity-server

move configuration to appsetings.json

JenBytowski commented on BookinV2.Data/BookingV2DataContext.cs:
```diff
@@ -0,0 +1,30 @@
+{
+    public class BookingV2DataContext : IBookinV2DataContext
+    {
+        private readonly BookingV2DBContext _db;
```
please add new line between field and constructor

JenBytowski commented on BookinV2.Data/BookingV2DataContext.cs:
```diff
@@ -0,0 +1,30 @@
+        private readonly BookingV2DBContext _db;
+        public BookingV2DataContext(BookingV2DBContext db)
+        {
+            _db = db;
```
would be great to add null check here as we discussed later

JenBytowski commented on BookinV2.Data/Entities/ApplicationUser.cs:
```diff
@@ -0,0 +1,8 @@
+
+namespace BookinV2.Data.Entities
+{
+    public class ApplicationUser : IdentityUser
```
as it is an entity then it should be inherited from BaseEntity

JenBytowski commented on BookinV2.Data/Entities/BaseEntity.cs:
```diff
@@ -0,0 +1,7 @@
+namespace BookinV2.Data.Entities
+{
+    public abstract class BaseEntity
```
you dont have to fix it now but you should start developing habit to write metainfo comments on important system parts like classes, public interface methods and so on

JenBytowski commented on BookinV2API/BookinV2API.csproj:
```diff
@@ -18,14 +18,42 @@
 
+	<ItemGroup>
+	  <Compile Remove="BookingV2Client\Новая папка\**" />
+	  <Compile Remove="Новая папка\**" />
```
do we really need this?)

JenBytowski commented on BookinV2API/BookinV2API.csproj.user:
```diff
@@ -0,0 +1,7 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
should be in the gitignore

JenBytowski commented on BookinV2API/Controllers/AuthController.cs:
```diff
@@ -0,0 +1,48 @@
+        [HttpPost("register")]
+        public async Task<IActionResult> Register([FromBody] RegisterModel model)
+        {
+            var user = new ApplicationUser { UserName = model.Username, Email = model.Email };
```
what would happen if model would be null?

JenBytowski commented on BookinV2API/Controllers/AuthController.cs:
```diff
@@ -0,0 +1,48 @@
+        [HttpPost("login")]
+        public async Task<IActionResult> Login([FromBody] LoginModel model)
+        {
+            var result = await this._signInManager.PasswordSignInAsync(model.Username, model.Password, false, false);
```
the same question as above here

JenBytowski replied:
also I think that we need to add common response model which would contain response meta info, like IsSucess flag, errors array and those kind of things to provide user a context in case something went wrong

JenBytowski commented on BookinV2API/Program.cs:
```diff
@@ -1,7 +1,55 @@
+
+builder.Services.AddScoped<IBookinV2DataContext, BookingV2DataContext>();
+
+var identityResources = builder.Configuration.GetSection("IdentityServer:IdentityResources").Get<List<IdentityResource>>();
```
it would be great to wrap it with extension method

JenBytowski commented on BookinV2API/Program.cs:
```diff
@@ -27,13 +75,26 @@
+// app.UseAuthentication();
+app.UseAuthorization();
+
+app.Use((y, x) =>
```
should be deleted

zloypelmen committed:
eliminate code review comments

add validation to auth controller, separate DB contexts and improve code readability

zloypelmen committed:
add BookinV2API.csproj.user to git ignore

zloypelmen committed:
Delete BookinV2API/BookinV2API.csproj.user

zloypelmen committed:
add register endpoint error handling

zloypelmen committed:
Merge branch 'feature/authorisation' of https://github.com/JenBytowski/BookingV2 into feature/authorisation

zloypelmen committed:
add errorMessages and errorResponse

zloypelmen committed:
add extention method

zloypelmen committed:
add unified ApiResponse format to authController

zloypelmen committed:
add discription to response

zloypelmen committed:
add new dbcontext extention method and improve api response
